server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPServer.java

```diff
@@ -75,8 +75,7 @@ private boolean wireTapEnabled() {
     private JMAPRoute.Action handleVersionRoute(HttpServerRequest request) {
         try {
             return jmapRoutesHandlers.stream()
-                .filter(jmapRoutesHandler -> jmapRoutesHandler.matches(request))
-                .flatMap(JMAPRoutesHandler::routes)
+                .flatMap(jmapRoutesHandler -> jmapRoutesHandler.routes(request))
                 .filter(jmapRoute -> jmapRoute.getEndpoint().matches(request))
```

Comment:
we don't really care to know if the Endpoint matches or if the jmapRoute matches, please write `jmapRoute.matches(request)` instead